rhesus-monkeys/scripts/convert-meta.py

```diff
@@ -0,0 +1,89 @@
+
+df_infant_w1 = df_meta_infant_bl.select(["Infant_ID", "Weight_PD7", "ActualDay_PD7"]).rename({"Weight_PD7": "weight_at_x_days_old", "ActualDay_PD7": "x_days_old"}).clean_names()
+df_infant_w2 = df_meta_infant_ur.select(["Infant_ID","Infant_weight","PD"]).rename({"Infant_weight": "weight_at_x_days_old", "PD": "x_days_old"}).clean_names()
+df_infant_weight = pl.concat([df_infant_w1, df_infant_w2]).unique().write_csv(folder_path / "data_infant_weight.csv", separator=";") # Concatenate (long)
+
+# Create the infant linking tables
+
+matter1 = "Blood"
+matter2 = "Urine"
+
+df_link_infant1 = df_meta_infant_bl.select(["Infant_ID", "Exp", "PD", "Batch"]).rename({"PD": "day_sample_taken"
+, "Exp": "infant_sample_id"}).clean_names().with_columns(pl.lit(matter1).alias("type_of_matter"))
+df_link_infant2 = df_meta_infant_ur.select(["Infant_ID", "Exp", "PD", "Batch"]).rename({"PD": "day_sample_taken"
+, "Exp": "infant_sample_id"}).clean_names().with_columns(pl.lit(matter2).alias("type_of_matter"))
+
+df_infant_link = pl.concat([df_link_infant1, df_link_infant2]).write_csv(folder_path / "data_infant_sample_meta.csv", separator=";") # Concatenate long
+
+# Create the adult metadata files
+    # FIO There are no additional Mother_ID in file 11
+df_metadata_maternal_bl = pl.read_csv(unzip_path / "metadata_maternal_blood.csv")
+df_metadata_maternal_ur = pl.read_csv(unzip_path / "metadata_maternal_urine.csv")
+df_metadata_maternal_pl = pl.read_csv(unzip_path / "metadata_maternal_placenta.csv")
+
+df_adult_meta1 = df_metadata_maternal_bl.select(["Mother_ID", "Mother_age", "GD_Delivery", "Group", "Mode_birth", "Reject", "Infant_sex"]).rename({"Mother_age": "Age_at_conception"
+, "GD_Delivery": "gestation_day_at_delivery", "Group": "obesity_classification", "Mode_birth": "mode_of_birth"}).clean_names()
+df_adult_meta2 = df_metadata_maternal_ur.select(["Mother_ID", "Mother_age", "GD_Delivery", "Group", "Mode_birth", "Reject", "Infant_sex"]).rename({"Mother_age": "Age_at_conception"
+, "GD_Delivery": "gestation_day_at_delivery", "Group": "obesity_classification", "Mode_birth": "mode_of_birth"}).clean_names()
+df_adult = pl.concat([df_adult_meta1, df_adult_meta2]).unique().write_csv(folder_path / "data_adult_meta.csv", separator=";") # Concatenate long
+
+df_adult_weight1 = df_metadata_maternal_bl.select(["Mother_ID", "GD_day", "GD_targeted", "Mother_Weight", "BCS"]).rename({"GD_day": "sample_gestation_day"
+, "GD_targeted": "target_gestation_day", "Mother_Weight": "weight_at_gestation_day", "BCS": "body_condition_score"})
+df_adult_weight2 = df_metadata_maternal_ur.select(["Mother_ID", "GD", "Target_GD", "Mother_Weight", "BCS"]).rename({"GD": "sample_gestation_day"
+, "Target_GD": "target_gestation_day", "Mother_Weight": "weight_at_gestation_day", "BCS": "body_condition_score"})
+df_adult_weight = pl.concat([df_adult_weight1, df_adult_weight2]).unique().write_csv(folder_path / "data_adult_weight.csv", separator=";") # Concatenate long
```

Comment:
Same as above

Reply:
@lwjohnst86 For the weight files the headers in the raw files are not named the same, so I can't do the rename after the concat, can I?

Reply:
Hmm, good point about the weight. Then instead, rather than select first, you can rename first, then concat, then select, since the variables being selected will be the same in both :relaxed:

Reply:
Or rather, `concat()`, then `clean_names()`, then `select()` then `unique()` etc.